src-tauri/src/main.rs

```diff
@@ -339,3 +367,26 @@ async fn get_node_edges(
 
     Ok(edges)
 }
+
+#[tauri::command]
+async fn run_algorithms(
+    bitfield: u8,
+    mesh_graph: tauri::State<'_, ActiveMeshGraph>,
+    algo_state: tauri::State<'_, ActiveMeshState>,
+) -> Result<AlgoStore, String> {
```

Comment:
Commands should return a `CommandState` error type, which implements `From<String>` so you might not need the `map_err` calls

Reply:
oh fair point. The `AlgoResults` also include `Error` and `Missing/None` options for each algo, should I unpack that here before I return to the frontend?

Reply:
What do you mean unpack them? All `CommandError` ([here](https://github.com/ajmcquilkin/meshtastic-emergency-response-client/blob/84c947b5cec2ff4acd91f6b59441630fbe14823d/src-tauri/src/main.rs#L27)) does is wrap the error message to support the JS Error spec, it doesn't have any special handling on its own

Reply:
my bad - that wasn't a directly relevant observation. It should have been its own comment. Agree with you on `CommandError`.

Reply:
Removed error mapping